kornia/augmentation/utils/helpers.py

```diff
@@ -82,10 +128,30 @@ def _validate_input_dtype(input: torch.Tensor, accepted_dtypes: List) -> None:
         raise TypeError(f"Expected input of {accepted_dtypes}. Got {input.dtype}")
 
 
-def _validate_shape(shape: Union[Tuple, torch.Size], required_shapes: List[str] = ["BCHW"]) -> None:
+def _transform_output_shape(output: Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]],
+                            shape: Tuple) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+    is_tuple = isinstance(output, tuple)
+    if is_tuple:
+        out_tensor, trans_matrix = output
+    else:
+        out_tensor = output
+        trans_matrix = None
+
+    for _ in range(len(out_tensor.shape) - len(shape)):
+        assert out_tensor.shape[0] == 1, f'Broadcasted dim has been changed to non-broadcasted dim.'
+        out_tensor = out_tensor.squeeze(0)
+
+    if trans_matrix is not None:
```

Comment:
Finish this as well. Probably only to unsqueeze if len(shape) == 2 .

Reply:
sorry I'm not sure I can follow. If input is `(H,W)`, what is the expected dimension of out trans matrix if `keepdim=False` and `keepdim=True`?

Reply:
Good question. I think if we are having ```keepdim```, we shall follow the same logic for both input tensor and transformation.

```python
if input_tensor is (C, H, W) or (H, W) and return_transform and keepdim:
    return (C, H, W) or (H, W), (3, 3)
if input_tensor is (C, H, W) or (H, W) and return_transform and not keepdim:
    return (1, C, H, W), (1, 3, 3)

if input_tensor is (B, C, H, W) and return_transform:
    return (B, C, H, W) and (B, 3, 3)
```

In order to avoid complicated scenarios, I think it is better to add a check to require both input_tensor and trans_mat are batched or unbatched.

/cc @edgarriba

Reply:
yes, something like that could work

Reply:
I made a check for output to check whether output and transformation matrix are in the same batching mode or not.